CodeStyleExercise/ColorsViewController.swift

```diff
@@ -8,88 +8,62 @@
 
 import UIKit
 
-class ColorsViewController : UIViewController, UICollectionViewDataSource, UITextFieldDelegate
-{
-
-    class constants {
-        static let DFLTNUMCOLORS:Int=12
-    }
-
-    @IBOutlet weak var collectionView:UICollectionView!
-    @IBOutlet weak var colorCountTextField: UITextField!
-
+final class ColorsViewController: UIViewController, UICollectionViewDataSource, UITextFieldDelegate {
+    
+    @IBOutlet private weak var collectionView: UICollectionView!
+    @IBOutlet private weak var colorCountTextField: UITextField!
+    
     var ct: Int? = 12
-
+    
+    class Constants {
+        static let defaultColorNumber = 12
+    }
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        self.collectionView.register(UINib(nibName: "ColorsCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "Cell")
-
-        self.colorCountTextField.placeholder = "12"
+        collectionView.register(UINib(nibName: "ColorsCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "Cell")
+        colorCountTextField.text = "\(Constants.defaultColorNumber)"
     }
-
-    override func didReceiveMemoryWarning() {
-        super.didReceiveMemoryWarning()
-    }
-
-    // MARK: - Collection View Data Source Methods
-
-    func collectionView(_ collectionView:UICollectionView, numberOfItemsInSection section : Int) -> Int{
+    
+    // MARK: - CollectionViewDataSource
+    func collectionView(_ collectionView:UICollectionView, numberOfItemsInSection section : Int) -> Int {
         return ct!
     }
-
+    
     func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-        var c = collectionView.dequeueReusableCell(withReuseIdentifier: "Cell", for: indexPath) as! ColorsCollectionViewCell
-//        let color = UIColor.random
-//        c.backgroundColor = color
-//        c.hexColorLbl.text = color.description
-        c.configureCellWithColor(color: random_ui_color())
+        let c = collectionView.dequeueReusableCell(withReuseIdentifier: "Cell", for: indexPath) as! ColorsCollectionViewCell
+        c.configureCell(withColor: random_ui_color())
         return c
     }
-
-
-
-
-    //////////////////////////////////
-    // MARK: - UITextFieldDelegate ///
-    //////////////////////////////////
-
+    
+    // MARK: - UITextFieldDelegate
     func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
```

Comment:
This method is a good candidate for refactoring using `guard` statements to avoid lots of nested `if`s (aka the pyramid of doom). See the solution code for an example of how to do this.